packages/swap/src/quoting/__tests__/QuoteRequest.test.ts

```diff
@@ -1,11 +1,65 @@
 /* eslint-disable dot-notation */
 import { beforeEach, describe, expect, it, vi } from 'vitest';
 import env from '../../config/env.js';
+import { getInternalSwapNetworkFeeInfo } from '../../polkadot/api.js';
 import { getUsdValue } from '../../pricing/checkPriceWarning.js';
+import { calculateRecommendedSlippage } from '../../utils/autoSlippage.js';
+import { isAtLeastSpecVersion } from '../../utils/function.js';
+import { getMinimumEgressAmount } from '../../utils/rpc.js';
+import { getSwapRateV3 } from '../../utils/statechain.js';
+import { estimateSwapDuration } from '../../utils/swap.js';
 import QuoteRequest, { MAX_NUMBER_OF_CHUNKS } from '../QuoteRequest.js';
 
 const originalEnv = structuredClone(env);
 
+vi.mock('../../utils/function', async (importOriginal) => {
+  const original = (await importOriginal()) as object;
+  return {
+    ...original,
+    isAtLeastSpecVersion: vi.fn().mockResolvedValue(true),
+  };
+});
+
+vi.mock('../../utils/statechain', async (importOriginal) => {
+  const original = (await importOriginal()) as object;
+  return {
+    ...original,
+    getSwapRateV3: vi.fn(),
+  };
+});
+
+vi.mock('../../utils/swap', async (importOriginal) => {
+  const original = (await importOriginal()) as object;
+  return {
+    ...original,
+    estimateSwapDuration: vi.fn(),
+  };
+});
+
+vi.mock('../../utils/autoSlippage', async (importOriginal) => {
+  const original = (await importOriginal()) as object;
+  return {
+    ...original,
+    calculateRecommendedSlippage: vi.fn(),
+  };
+});
+
+vi.mock('../../utils/rpc', async (importOriginal) => {
+  const original = (await importOriginal()) as object;
+  return {
+    ...original,
+    getMinimumEgressAmount: vi.fn(),
+  };
+});
+
+vi.mock('../../polkadot/api', async (importOriginal) => {
+  const original = (await importOriginal()) as object;
+  return {
+    ...original,
+    getInternalSwapNetworkFeeInfo: vi.fn(),
+  };
+});
```

Comment:
most of these can probably use a simple mock:
```suggestion
vi.mock('../../polkadot/api');
```